File: forge/src/main/java/militarymod/tabs/MilitaryTabs.java
package militarymod.tabs;

import militarymod.MilitaryMod;
import militarymod.blocks.ModBlocks;
import net.minecraft.network.chat.Component;
import net.minecraft.network.chat.TranslatableComponent;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.block.Blocks;

public class MilitaryTabs {

    public static CreativeModeTab TEST_TAB = new CreativeModeTab("test_tab") {
        @Override
        public ItemStack makeIcon() {
            return new ItemStack(Blocks.IRON_BLOCK.asItem());
        }

        @Override
        public Component getDisplayName() {
            return new TranslatableComponent("itemGroup." + String.format("%s.%s", MilitaryMod.modId, "test_tab"));
        }
    };

}

File: forge/src/main/kotlin/militarymod/tabs/MilitaryTabs.kt
package militarymod.tabs

import militarymod.MilitaryMod
import net.minecraft.world.item.CreativeModeTab
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.block.Blocks

object MilitaryTabs {

    private val tk = MilitaryMod.MODID+"."

    val TAB = object : CreativeModeTab(tk+"tab") {
        override fun makeIcon(): ItemStack {
            return ItemStack(Blocks.IRON_BLOCK)
        }
    }

}